src/test/java/com/puppycrawl/tools/checkstyle/checks/naming/AbbreviationAsWordInNameCheckTest.java

```diff
@@ -226,12 +227,169 @@ public void testTypeAndVariablesAndMethodNamesWithIgnoresStatic() throws Excepti
             + ",METHOD_DEF,ENUM_DEF,ENUM_CONSTANT_DEF"
             + ",PARAMETER_DEF,INTERFACE_DEF,ANNOTATION_DEF");
         checkConfig.addAttribute("ignoreOverriddenMethods", "true");
-        final int expectedCapitalCount = 6;
+        final int expectedCapitalCount = 5;
+
+        final String[] expected = {
+            "12:16: " + getWarningMessage("AbstractCLASSName", expectedCapitalCount),
+            "32:11: " + getWarningMessage("AbstractINNERRClass", expectedCapitalCount),
+            "37:11: " + getWarningMessage("WellNamedFACTORY", expectedCapitalCount),
+            "38:21: " + getWarningMessage("marazmaticMETHODName", expectedCapitalCount),
+            "58:20: "
+                + getWarningMessage("serialNUMBER", expectedCapitalCount), // not in ignore list
+            "59:26: "
+                + getWarningMessage("s1erialNUMBER", expectedCapitalCount), // no ignore for final
+        };
+
+        verify(checkConfig, getPath("InputAbbreviationAsWordInNameType.java"), expected);
+    }
+
+    @Test
+    public void testTypeAndVariablesAndMethodNamesWithIgnoresStaticFinal() throws Exception {
```

Comment:
please make separate test inputs for each configuration with configuration explanation comment before type definition, e.g.
```
/* 
* Config:
*  prop1 = true
* prop2 = false
*/
public class MyTestInput {}   
```

. It will be easier to understand and to mantain it.

Reply:
Added.

Would appreciate a review on whether the comments are consistent with the individual UTs, they _should_ be mostly identical but it's hard to keep track of all 8 cases.